clibrary.h

```diff
@@ -1,12 +1,13 @@
 #ifndef CLIBRARY_H
 #define CLIBRARY_H
 
-#include <tiledb/tiledb.h>
+#include <tiledb/tiledb_experimental.h>
```

Comment:
We could create a `vfs_experimental.go` and `clibrary_experimental.{h,c}` to keep non experimental and experimental features separate

Reply:
We could, but I'm not sure what the benefit would be. We can add a `//go:build experimental` at the top of the file like [this file](https://github.com/TileDB-Inc/TileDB-Go/pull/225/files#diff-0d15a00563eb8bca6ddb7ef55eb18247159f8419c17be402ab23d65330d319d9) in #225 does, but currently we are not doing this in the existing `experimental` files.